File: articles/claude-shannon-and-the-mathematical-theory-of-information/typescript/entropy_example.ts
function entropy(probs: number[]): number {
  return -probs
    .filter((p) => p > 0)
    .map((p) => p * Math.log2(p))
    .reduce((a, b) => a + b, 0);
}

console.log(`entropy_bits=${entropy([0.5, 0.5]).toFixed(9)}`);

File: articles/claude-shannon-and-the-mathematical-theory-of-information/rust/entropy_example.rs
fn entropy(probs: &[f64]) -> f64 {
    -probs.iter().filter(|p| **p > 0.0).map(|p| p * p.log2()).sum::<f64>()
}

fn main() {
    println!("entropy_bits={:.9}", entropy(&[0.5, 0.5]));
}
